### PR TITLE
Storage update

### PR DESCRIPTION
**Pull request by anuchandy**
## Description

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [ ] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md).**
- [ ] **The pull request does not introduce [breaking changes](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/Documentation/breaking-changes.md).**

### [General Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#general-guidelines)
- [ ] Title of the pull request is clear and informative.
- [ ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/dev/documentation/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#testing-guidelines)
- [ ] Pull request includes test coverage for the included changes.

### [SDK Generation Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#sdk-generation-guidelines)
- [ ] If an SDK is being regenerated based on a new swagger spec, a link to the pull request containing these swagger spec changes has been included above.
- [ ] The generate.cmd file for the SDK has been updated with the version of AutoRest, as well as the commitid of your swagger spec or link to the swagger spec, used to generate the code.
- [ ] The `*.csproj` and `AssemblyInfo.cs` files have been updated with the new version of the SDK.

### CODE REVIEW

**Comment by anuchandy** [comment]
1. Fluent storage was completely handwritten. This PR contains regenerated interfaces and implementation.
2. Adding support for encryption.
3. Adding more test cases.

File: src/ResourceManagement/Storage/StorageAccountImpl.cs [hunk 1]
```diff
@@ -0,0 +1,376 @@
...
+        {
+            // TODO: In the next major version upgrade of SDK change return type to nullable
+            //       returning the default enum value when server returns null is not expected
+            return Inner.AccessTier.HasValue ? Inner.AccessTier.Value : default(AccessTier);
+        }
+
+        ///GENMHASH:E4DFA7EA15F8324FB60C810D0C96D2FF:1BCD5CF569F11AB6F798D4F3A5BFC786
+        public IReadOnlyList<Models.StorageAccountKey> GetKeys()
+        {
+            return GetKeysAsync().ConfigureAwait(false).GetAwaiter().GetResult();
```
Comment (hovsepm) [inline src/ResourceManagement/Storage/StorageAccountImpl.cs]: Please change all ".ConfigureAwait(false).GetAwaiter().GetResult();" to Extensions.Synchronize(() => <your ASYNC method call>)

File: src/ResourceManagement/Storage/StorageAccountImpl.cs [hunk 2]
```diff
@@ -0,0 +1,376 @@
...
+            var keysResultInner = await this.Manager.Inner.StorageAccounts.ListKeysAsync(this.ResourceGroupName, this.Name, cancellationToken);
+            var keys = new List<StorageAccountKey>();
+            keys.AddRange(keysResultInner.Keys);
+            return keys;
+        }
+
+        ///GENMHASH:FE5C90217FF36474FA8DE7E91403E40F:8A9D1B7CB45D0ABAC76D65E99FADA580
+        public IReadOnlyList<Models.StorageAccountKey> RegenerateKey(string keyName)
+        {
+            return RegenerateKeyAsync(keyName).ConfigureAwait(false).GetAwaiter().GetResult();
```
Comment (hovsepm) [inline src/ResourceManagement/Storage/StorageAccountImpl.cs]: here as well use  Extensions.Synchronize(()  instead of GetAwaiter.

**Review by hovsepm — CHANGES_REQUESTED** [review]

File: src/ResourceManagement/Storage/StorageAccountImpl.cs (hunk 1 above)
Comment (anuchandy) [inline src/ResourceManagement/Storage/StorageAccountImpl.cs]: good catch,thanks!

**Review by hovsepm — APPROVED** [review]

**Pull request merged**